15-languages.Rmd

```diff
@@ -273,9 +273,17 @@ The `stata` engine in **knitr** is quite limited. Doug Hemken has substantially
 
 ## Create graphics with Asymptote {#eng-asy}
 
+```{r echo = FALSE, warning = FALSE}
+check_not_windows <- (Sys.info()['sysname'] != 'Windows')
+check_has_asymptote <- (system("which asy", intern = TRUE) == 0)
+check_ci <- is.na(Sys.getenv('CI', NA))
+eval_asy <- (check_not_windows & check_has_asymptote & check_ci)
+if (!eval_asy) warning("System set-up not compatible with Asymptote, so chunks with asy engine will be skipped.")
+```
```

Comment:
What I meant by function was the same as that but to be able to use the function in `eval` later

```suggestion
```{r include = FALSE}
eval_asy <- function() {
	check_not_windows <- Sys.info()['sysname'] != 'Windows'
	check_has_asymptote <- nzchar(Sys.which("asy"))
	eval_asy <- check_not_windows & check_has_asymptote
	if (!eval_asy) 
		warning("System set-up not compatible with Asymptote, so chunks with asy engine will be skipped.")
	eval_asy
}
```
```

That way we could use `eval_asy()` in the chunk option and get the warning.